src/odemis/driver/hamamatsurx.py

```diff
@@ -0,0 +1,2126 @@
+
+    def _setResolution(self, _=None):
+        """
+        Sets the resolution VA.
+        So far the full field of view is always used. Therefore, resolution only changes with binning.
+        :return: current resolution value
+        """
+        # Note: we can keep it simple as long as we do not provide to change the sensor size yet...
+        resolution = self._shape[:2]
+        new_res = (int(resolution[0] // self._binning[0]),
+                   int(resolution[1] // self._binning[1]))  # floor division
+
+        if self._spectrograph:
+            self._updateWavelengthList()  # update WavelengthList when changing binning
+
+        return new_res
+
+    def _getCamExpTimeRange(self):
+        """
+        Get min and max values for the camera exposure time.
+        :return: tuple containing min and max exposure time
+        """
+        exp = self.parent.CamParamInfoEx("Live", "Exposure")  # returns list
+        # Values in returned list "exp" are in order. 1st - 4th values see CamParamInfoEx.
+        min_value = exp[4]
+        max_value = exp[-1]
+
+        min_value_raw, min_unit = min_value.split(' ')[0:2]
+        max_value_raw, max_unit = max_value.split(' ')[0:2]
+
+        min_exp = self.parent.convertUnit2Time(min_value_raw, min_unit)
+        max_exp = self.parent.convertUnit2Time(max_value_raw, max_unit)
+
+        return min_exp, max_exp
+
+    def GetCamExpTime(self):
+        """
+        Get the camera exposure time.
+        Converts the provided value received from RemoteEx into sec.
+        :return: (float) exposure time in sec
+        """
+        exp_time_raw = self.parent.CamParamGet("Live", "Exposure")[0].split(' ')
+        try:
+            exp_time = self.parent.convertUnit2Time(exp_time_raw[0], exp_time_raw[1])
+        except Exception:
+            raise logging.error("Exposure time of %s is not supported for read-out camera." % exp_time_raw)
+
+        return exp_time
+
+    def _setCamExpTime(self, value):
+        """
+        Set the camera exposure time.
+        Converts the time range in sec into a for RemoteEx readable format.
+        :parameter value: (float) exposure time to be set
+        :return: (float) current exposure time
+        """
+        try:
+            exp_time_raw = self.parent.convertTime2Unit(value)
+        except Exception:
+            raise logging.debug("Exposure time of %s sec is not supported for read-out camera." % value)
+
+        # Note: RemoteEx uses different exposure times depending on acquisition mode
+        # If we support e.g. photon counting, we need to specify a different location in RemoteEx.
+        # For now location is always "Live"
+        self.parent.CamParamSet("Live", "Exposure", exp_time_raw)
+        self._metadata[model.MD_EXP_TIME] = value  # update MD
+
+        return value
+
+    def _start(self):
+        """
+        Start an acquisition.
+        :raises CancelledError if the acquisition must stop.  # TODO
```

Comment:
I don't think it's useful/meaningful to raise such error. The caller will subscribe/unsubscibe to the dataflow. If there is at least one subscriber the acquisition should be active, otherwise it should be stopped. There is no reason that subscribing would raise a CancelledError.

Reply:
So completely remove CancelledError in general?

Reply:
Yes